src/events/guild/message.js

```diff
@@ -0,0 +1,31 @@
+const { prefix } = require("../../../config.json")
+const ConfigModel = require("../../database/schemas/config")
+
+module.exports = async (client, message) => {
+    let args = message.content.slice(prefix.length).trim().split(/ +/g);
+    let cmd = args.shift().toLowerCase();
+    if (message.author.bot) return;
+
+    for (i = 0; i < args.length; i++)
+        if (args[i] === '||')
+            args = args.slice(0, i);
+
+    if (message.content.startsWith(prefix)) {
+        let commandfile = client.commands.get(cmd) || client.commands.get(client.aliases.get(cmd))
+        if (commandfile) { 
+            const authRoles = message.member.roles.cache;
+            if (commandfile.config.accessibility === "Member")
+                commandfile.run(client, message, args)
+            if (commandfile.config.accessibility === "Moderator") {
```

Comment:
I would make this a `switch` statement as the accessibility can only be one value.
Also all my comments about adding an `Administrator` option for some commands.

Reply:
Changed it to the `switch`, though I preferred the `if` statements due to the braces. Added `Administrator` option for command accessibility too.

Reply:
A switch statement can still use braces if you prefer them, some times they are needed when variables of the same name are used within different cases. However, I believe the break needs to be outside the braces, but I would need to check.
A neat trick you can do with cases if to have one lead into the next one. ie. if the check for "Moderator" fails it moves onto the check for "Administrator" although very minor it is a interesting thing to consider.
You don't need to add this, but here is what it looks like:
```js
switch (commandfile.config.accessibility) {
    case "Member":
        commandfile.run(client, message, args)
        break
    case "Moderator":
        const config = await ConfigModel.findOne({guildid: message.guild.id})
        if (authRoles.some((r) => r.id === config.moderatorroleId)) {
            commandfile.run(client, message, args)
            break
        }
    case "Administrator":
        if (message.guild.member(message.author).hasPermission('ADMINISTRATOR'))
            commandfile.run(client, message, args)
        else
            message.reply("Insufficient permissions :P")
        break
    default:
        message.reply("Wrong accessibility. Command not run")
```